app/data/ops.py

```diff
@@ -0,0 +1,101 @@
+import os
+
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+
+from app.config.parser import ConfigParser
+from app.data.ids import create_id
+from app.data.models import Image, Label, Base
+
+
+class ImageDataHandler:
+
+    def __init__(self):
+        Config = ConfigParser()
+
+        self.session = create_session(Config.data_file())
```

Comment:
session is created but never closed

https://docs.sqlalchemy.org/en/13/orm/session_api.html#sqlalchemy.orm.session.Session.close

Reply:
I fixed this issue by implementing an auto_session method which automatically closes the session. See the changes by the new commit.